nodes/nomos-node/node/src/main.rs

```diff
@@ -1,69 +1,32 @@
 use std::collections::HashMap;
 
-use clap::Parser;
+use clap::Parser as _;
 use color_eyre::eyre::{eyre, Result};
 use kzgrs_backend::dispersal::BlobInfo;
 use nomos_core::{da::blob::info::DispersedBlobInfo, tx::Transaction};
 use nomos_membership::{backends::mock::MockMembershipBackendSettings, BackendSettings};
 use nomos_mempool::{
     network::adapters::libp2p::Settings as AdapterSettings, tx::settings::TxMempoolSettings,
 };
-use nomos_node::{
-    config::BlendArgs, Config, CryptarchiaArgs, HttpArgs, LogArgs, NetworkArgs, Nomos,
-    NomosServiceSettings, Tx,
-};
-use overwatch::overwatch::OverwatchRunner;
-
-#[derive(Parser, Debug)]
-#[command(author, version, about, long_about = None)]
-struct Args {
-    /// Path for a yaml-encoded network config file
-    config: std::path::PathBuf,
-    /// Dry-run flag. If active, the binary will try to deserialize the config
-    /// file and then exit.
-    #[clap(long = "check-config", action)]
-    check_config_only: bool,
-    /// Overrides log config.
-    #[clap(flatten)]
-    log: LogArgs,
-    /// Overrides network config.
-    #[clap(flatten)]
-    network: NetworkArgs,
-    /// Overrides blend config.
-    #[clap(flatten)]
-    blend: BlendArgs,
-    /// Overrides http config.
-    #[clap(flatten)]
-    http: HttpArgs,
-    #[clap(flatten)]
-    cryptarchia: CryptarchiaArgs,
-}
+use nomos_node::{config::CliArgs, Config, Nomos, NomosServiceSettings, RuntimeServiceId, Tx};
+use overwatch::overwatch::{Error as OverwatchError, Overwatch, OverwatchRunner};
 
 #[tokio::main]
 async fn main() -> Result<()> {
-    let Args {
-        config,
-        check_config_only,
-        log: log_args,
-        http: http_args,
-        network: network_args,
-        blend: blend_args,
-        cryptarchia: cryptarchia_args,
-    } = Args::parse();
-    let config = serde_yaml::from_reader::<_, Config>(std::fs::File::open(config)?)?
-        .update_from_args(
-            log_args,
-            network_args,
-            blend_args,
-            http_args,
-            cryptarchia_args,
-        )?;
+    let cli_args = CliArgs::parse();
+    let is_dry_run = cli_args.dry_run();
+    let must_blend_service_group_start = cli_args.must_blend_service_group_start();
+    let must_da_service_group_start = cli_args.must_da_service_group_start();
+
+    let config =
+        serde_yaml::from_reader::<_, Config>(std::fs::File::open(cli_args.config_path())?)?
+            .update_from_args(cli_args)?;
 
     #[expect(
         clippy::non_ascii_literal,
         reason = "Use of green checkmark for better UX."
     )]
-    if check_config_only {
```

Comment:
I think this slipped through in my rebase. I don't see any need to change the naming of the variable.

Reply:
This was renamed to `dry_run` in this PR, but on `master` it's `check_config_only`. Up to you whatever name you prefer, but I was pointing out this PR changes it, not intentionally.